### PR TITLE
[DOCS] Qualified Watcher topics to distinguish from Kib Alerting

### PR DESCRIPTION
**Pull request by debadair**
With the introduction of Kibana Alerting, using Alerting in the Watcher docs is confusing, as are the unqualified actions that overlap. The overlap also has SEO implications that make it less likely for folks to find the Kibana Alerting docs. 

This PR qualifies the Watcher headings and anchors that generate the HTML page names, adds the old pages to the redirects file, and updates the necessary xrefs.

### CODE REVIEW

**Comment by jrodewig** [comment]
I wonder if simply changing the headings here would suffice.

Changing anchors requires a ton of overhead and creates an awful experience for users with the old URL/anchor. While I know the URL impacts SEO, I wonder if the marginal gains are worth the friction for users.

**Comment by debadair** [comment]
@elasticmachine retest this please

**Comment by debadair** [comment]
I had to revert some of the anchor changes, so ended up reverting all of them. It means that these pages are still possibly more likely to surface in search results than the Kibana Alerting pages, but the heading changes should help clarify things and this way it isn't as disruptive for folks who are using Watcher.

File: x-pack/docs/en/watcher/trigger.asciidoc [hunk 1]
```diff
@@ -1,6 +1,9 @@
 [role="xpack"]
-[[trigger]]
-== Triggers
+[[triggers]]
```
Comment (jrodewig) [inline x-pack/docs/en/watcher/trigger.asciidoc:2]: Not sure if this anchor change was intentional. If so, we should add a redirect for it.

File: x-pack/docs/en/watcher/actions/email.asciidoc [hunk 2]
```diff
@@ -1,6 +1,9 @@
 [role="xpack"]
 [[actions-email]]
-=== Email action
+=== Watcher email action
```
Comment (jrodewig) [inline x-pack/docs/en/watcher/actions/email.asciidoc:3]: Should we use the attribute here?
```suggestion
=== {watcher} email action
```

File: x-pack/docs/en/watcher/actions/index.asciidoc [hunk 3]
```diff
@@ -1,6 +1,9 @@
 [role="xpack"]
 [[actions-index]]
-=== Index action
+=== Watcher index action
```
Comment (jrodewig) [inline x-pack/docs/en/watcher/actions/index.asciidoc:3]:
```suggestion
=== {watcher} index action
```

File: x-pack/docs/en/watcher/gs-index.asciidoc [hunk 4]
```diff
@@ -1,6 +1,6 @@
 [role="xpack"]
-[[xpack-alerting]]
-= Alerting on cluster and index events
+[[watcher-alerting]]
+= Watching for cluster and index events
```
Comment (jrodewig) [inline x-pack/docs/en/watcher/gs-index.asciidoc:3]: This file looks inert. Maybe remove it entirely?

Also not sure why this isn't just `Watch for...`

File: x-pack/docs/en/watcher/release-notes.asciidoc [hunk 5]
```diff
@@ -205,7 +205,7 @@ October 28, 2015
   use the following index name `<logstash-{now/d{YYYY.MM.dd|+01:00}}>`.
 
 .New Features
-* Added new <<actions-hipchat,HipChat Action>>
+* Added new <<actionss-hipchat,HipChat Action>>
```
Comment (jrodewig) [inline x-pack/docs/en/watcher/release-notes.asciidoc:208]: Looks like a typo. I think this file may be inert.

**Review by jrodewig — APPROVED** [review]
LGTM. However, I think we may be missing a redirect for one remaining anchor change.

File: x-pack/docs/en/watcher/index.asciidoc [hunk 6]
```diff
@@ -1,11 +1,16 @@
...
 [[xpack-alerting]]
-= Alerting on cluster and index events
+= Watch for cluster and index events
 
 [partintro]
 --
-The {alert-features} enable you to watch
-for changes or anomalies in your data and perform the necessary actions in
+TIP: {kib} Alerting provides a set of built-in actions and alerts 
+that are integrated with the APM, Metrics, Security, and Uptime applications.
```
Comment (gchaps) [inline x-pack/docs/en/watcher/index.asciidoc:8]: There will be more apps that integrate with alerting.

**Review by gchaps — APPROVED** [review]
LGTM

File: x-pack/docs/en/watcher/trigger.asciidoc (hunk 1 above)
Comment (debadair) [reply x-pack/docs/en/watcher/trigger.asciidoc:2]: I'll revert that one, too.

File: x-pack/docs/en/watcher/gs-index.asciidoc (hunk 4 above)
Comment (debadair) [reply x-pack/docs/en/watcher/gs-index.asciidoc:3]: Will remove it. It's an artifact from when we built standalone getting started "books" for the translation effort.

**Pull request merged**